### DISCUSSION TITLE
How do I make dummy proofs with snarky/pickles?

### DISCUSSION THREAD

**Discussion by L-as** [MinaProtocol/mina#14464] [Q&A] [answered]
I want to test my circuits and make sure they're correct without making real proofs. Some parts of the code reference a "proof level", but I wasn't able to find anything that does what I want.

**Comment by mrmr1993** [comment, marked as answer]
Pickles doesn't support that. You can see the use of the protocol-level flag [here](https://github.com/MinaProtocol/mina/blob/85bc56cc4a54a4fb94dd06af3adc1c3b088ba021/src/lib/prover/prover.ml#L88) and [here](https://github.com/MinaProtocol/mina/blob/85bc56cc4a54a4fb94dd06af3adc1c3b088ba021/src/lib/snark_worker/prod.ml#L72).

You can use snarky's `check`, similar to how the prover does [here](https://github.com/MinaProtocol/mina/blob/85bc56cc4a54a4fb94dd06af3adc1c3b088ba021/src/lib/prover/prover.ml#L174), via [this call](https://github.com/MinaProtocol/mina/blob/85bc56cc4a54a4fb94dd06af3adc1c3b088ba021/src/lib/blockchain_snark/blockchain_snark_state.ml#L402) if you want to test the non-recursive aspects of your circuit.

**Reply by L-as** [reply]
This is very unfortunate!
I want to implement a dummy backend that bypass kimchi and does it the following way:
- A proof is a hash of the circuit and the public inputs.
- An invalid proof is anything else.

This is obviously insecure, but it seems to me like it would be good enough for testing, if you trust that no one generates malicious proofs (which should be the case for most tests).

I'm still in the process of understanding the architecture, but I imagine I'd have to rewrite kimchi_backend? Or is there somewhere else that's more apt? I could do it directly in the kimchi crate too but I prefer not going through the FFI for testing if possible.
